swri_roscpp/readme.md

```diff
@@ -31,3 +31,51 @@ add_topic_service_files(DIRECTORY topic_srv FILES
 ```
 
 To use the generated topics simply include `your_project/your_service.h`. They have response and request fields just like normal services.
+
+## Dynamic Parameters
+
+A class that implements the same functionality as Dynamic Reconfigure, but in a more dynamic and easier to use way. Instead of declaring configurable parameters at compile time in a config file they are specified in your C++ code.
+
+To use, add the class to your node and call the initalize function with a NodeHandle in your private namespace.
+
+```cpp
+#include <swri_roscpp/dynamic_parameters.h>
+
+void main()
+{
+  // your node stuff here
+
+  ros::NodeHandle pnh("~");
+  swri::DynamicParameters params;
+  params.initialize(pnh);
+...
+```
+
+You can then start declaring and reading in configuration values, giving a long living pointer to a variable for each. This variable is changed on a dynamic reconfigure of that parameter. If you want to ensure no changes are applied during an operation, be sure to lock the mutex you can get by calling the `params.mutex()` function.
```

Comment:
I think I'd prefer this to say that you should *always* lock the mutex around getting the parameters, and also include and example of that below.  There are some cases where you can get away without locking it, but I'd rather be safe than sorry and leave figuring that out to people who really know what they're doing.